pykilosort/gui/data_view_box.py

```diff
@@ -499,14 +521,103 @@ def generate_lookup_table(self, colormap_min, colormap_max, num_points=8192):
 
-    def add_curve_to_plot(self, trace, color, label):
-        curve = pg.PlotCurveItem(
-            parent=self.plot_item, clickable=True, pen=pg.mkPen(color=color, width=1)
-        )
+    def create_plot_items(self):
+        """
+        Create curve plot items for each active view.
 
-        curve.label = label
-        curve.setData(trace * self.scale_factor + 200 * label)
-        self.plot_item.addItem(curve)
+        Loops over all views and creates curve plot items for each view.
+        Creating plot items beforehand results in quicker plotting and a
+        smoother GUI experience when scrolling through data in traces
+        mode.
+        """
+
+        for view in self._keys:
+            self.traces_plot_items[view] = []
+            for c, channel in enumerate(
+                    range(
+                        self.primary_channel + self.channels_displayed_traces,
+                        self.primary_channel,
+                        -1
+                    )
+            ):
+                curve = pg.PlotCurveItem()
+                curve.setPos(0, 200 * c)
+                self.plot_item.addItem(curve)
+
+                self.traces_plot_items[view].append(curve)
+
+    def add_traces_to_plot_items(
+            self,
+            traces: np.ndarray,
+            good_channels: t.Union[list, np.ndarray],
+            view: str
+    ):
+        """
+        Update plot items with traces.
+        
+        Loops over traces and plots each trace using the setData() method
+        of pyqtgraph's PlotCurveItem. The color of the trace depends on
+        the mode requested (raw, whitened, prediction, residual). Bad
+        channels are plotted in a different color. Each trace is also
+        scaled by a certain factor defined in self.traces_scaling_factor.
+        
+        Parameters
+        ----------
+        traces : numpy.ndarray
+            Data to be plotted.
+        good_channels : list or numpy.ndarray
+            Iterable with boolean values for good (True) or bad (False)
+             channels, determined during the preprocessing step
+        view : str
+            One of "raw", "whitened", "prediction" and "residual" views
+        """
+        for c, channel in enumerate(
+                range(
+                    self.primary_channel,
+                    self.primary_channel + self.channels_displayed_traces,
+                )
+        ):
+            try:
+                curve = self.traces_plot_items[view][c]
+                color = (
+                    self.traces_curve_color[view]
+                    if good_channels[channel]
+                    else self.bad_channel_color
+                )
+                curve.setPen(color=color, width=1)
+                curve.setData(
+                    traces.T[channel] *
+                    self.scale_factor *
+                    self.traces_scaling_factor[view]
+                )
+            except IndexError:
```

Comment:
suggestion: add a comment to note when this happens, if its not intended to happen then add a warning message